r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -386,6 +389,7 @@ test_that("extract epiweek from timestamp", {
 })
 
 test_that("extract week from timestamp", {
+  skip_on_os("windows") # https://issues.apache.org/jira/browse/ARROW-13168
```

Comment:
FYI: the error I saw in one of the logs for this one looks subtly different from the others:

https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38086

```
-- Error (test-dplyr-funcs-datetime.R:390:3): extract week from timestamp ------
[38087](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38087)
Error in `handle_csv_read_error(e, x$.data$schema)`: Invalid: Cannot locate timezone 'UTC': Timezone database not found at "C:\Users\runneradmin\Downloads\tzdata"
[38088](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38088)
Backtrace:
[38089](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38089)
     x
[38090](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38090)
  1. +-arrow:::compare_dplyr_binding(...) at test-dplyr-funcs-datetime.R:390:2
[38091](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38091)
  2. | +-testthat::expect_warning(...) at D:\a\arrow\arrow\r\check\arrow.Rcheck\tests_x64\testthat\helper-expectation.R:108:4
[38092](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38092)
  3. | | \-testthat:::expect_condition_matching(...)
[38093](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38093)
  4. | |   \-testthat:::quasi_capture(...)
[38094](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38094)
  5. | |     +-testthat .capture(...)
[38095](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38095)
  6. | |     | \-base::withCallingHandlers(...)
[38096](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38096)
  7. | |     \-rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
[38097](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38097)
  8. | \-rlang::eval_tidy(expr, rlang::new_data_mask(rlang::env(.input = record_batch(tbl))))
[38098](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38098)
  9. +-.input %>% mutate(x = week(datetime)) %>% collect()
[38099](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38099)
 10. +-dplyr::collect(.)
[38100](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38100)
 11. \-arrow:::collect.arrow_dplyr_query(.)
[38101](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38101)
 12.   \-base::tryCatch(...)
[38102](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38102)
 13.     \-base tryCatchList(expr, classes, parentenv, handlers)
[38103](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38103)
 14.       \-base tryCatchOne(expr, names, parentenv, handlers[[1L]])
[38104](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38104)
 15.         \-value[[3L]](cond)
[38105](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38105)
 16.           \-arrow:::handle_csv_read_error(e, x$.data$schema)
[38106](https://github.com/apache/arrow/runs/5067978610?check_suite_focus=true#step:18:38106)
 17.             \-rlang::abort(msg)
```

That `handle_csv_read_error()` has a bit of a funny name, but _is_ indeed called during `collect()`: https://github.com/apache/arrow/blob/47d55752172af99fce629f8fe6177df6b41af1d3/r/R/dplyr-collect.R#L33


Are we sure that this error is happening only in the testing code (e.g. when we try and compare the dates they are subtly different cause the timezone database doesn't exist / we can't compare them cause the timezone database doesn't exist?), or is this error happening in `collect()` which would be a problem for windows users attempting to use this function?

Reply:
@jonkeane Do you think we could move this forward before [ARROW-13168](https://issues.apache.org/jira/browse/ARROW-13168) is solved?

Reply:
Yes, though I do think we need to figure out what's going on in the case I highlighted in this comment thread. Have you figured out what's going on here already (I looked for a comment / explanation and haven't seen any but I might very well have missed it, so if you have please link me to it!)?

Reply:
https://github.com/apache/arrow/pull/12240#issuecomment-1047822854 (we were writing at the same time)